central/sensor/service/connection/upgradecontroller/active.go

```diff
@@ -39,18 +38,20 @@ func (u *upgradeController) makeProcessActive(cluster *storage.Cluster, processS
 
 func (u *upgradeController) maybeTimeoutUpgrade(processID string) error {
 	currState := u.active.status.GetProgress().GetUpgradeState()
-	var relevantTimestamp *types.Timestamp
-	if currState == storage.UpgradeProgress_UPGRADE_INITIALIZING {
-		relevantTimestamp = u.active.status.GetInitiatedAt()
-	}
-	if relevantTimestamp == nil {
-		relevantTimestamp = u.active.status.GetProgress().GetSince()
-	}
-	if relevantTimestamp == nil {
+	var relevantGoTime time.Time
+	if currState == storage.UpgradeProgress_UPGRADE_INITIALIZING &&
+		u.active != nil &&
+		u.active.status != nil &&
```

Comment:
I would suggest pulling these checks before ifs.

Maybe a few first lines of this function could be:
```
	if u == nil || u.active == nil || u.active.status == nil {
		return errors.Errorf("got no relevant timestamp for upgrade controller with status: %+v", u.upgradeStatus)
	}
```